File: Leetcode75/1456_Max_num_vowels.py
class Solution(object):
    def maxVowels(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: int
        """
        vowels = "aeiou"
        maxValue = 0
        for i in range(k):
            if s[i] in vowels:
                maxValue += 1

        window = s[:k]
        curValue = maxValue
        for i in range(len(s) - k):
            if s[i] in vowels:
                curValue -= 1
            if s[i+k] in vowels:
                curValue += 1
            if curValue > maxValue:
                maxValue = curValue
        return maxValue
        

File: Leetcode75/1456_max_num_vowels.java
class Solution {
    private static boolean isVowel(char c) {
        if (c == 'a' || c == 'e' ||c == 'i' ||c == 'o' ||c == 'u') {
            return true;
        }
        return false;
    }
    public int maxVowels(String s, int k) {
        int maxValue = 0;
        for (int i = 0 ; i < k ; i ++) {
            if (isVowel(s.charAt(i))) {
                maxValue++;
            }
        }
        int curValue = maxValue;
        for (int i = 0 ; i < s.length() - k ; i ++) {
            if (isVowel(s.charAt(i))) {
                curValue--;
            }
            if (isVowel(s.charAt(i+k))) {
                curValue++;
            }
            if (curValue > maxValue) {
                maxValue = curValue;
            }
        }
        return maxValue;
    }
}
